anitya/tests/db/test_models.py

```diff
@@ -175,9 +208,164 @@ def test_project_get_or_create(self):
             backend='foobar'
         )
 
+    def test_project_get_or_create_get(self):
+        """
+        Assert that project is only returned when existing name
+        is provided.
+        """
+        create_project(self.session)
+        pre_projects = models.Project.search(self.session, '*', count=True)
+        project = models.Project.get_or_create(
+            self.session,
+            name='geany',
+            homepage='https://www.geany.org/')
+        post_projects = models.Project.search(self.session, '*', count=True)
+        self.assertEqual(project.name, 'geany')
+        self.assertEqual(project.homepage, 'https://www.geany.org/')
+        self.assertEqual(pre_projects, post_projects)
+
+    def test_project_updated_new(self):
+        """
+        Assert that only new projects are returned when status is
+        set to 'new'.
+        """
+        create_project(self.session)
+
+        projects = models.Project.updated(self.session, status='new')
+        self.assertEqual(len(projects), 3)
+        self.assertEqual(projects[0].logs, None)
+
+    def test_project_updated_newer_updated(self):
+        """
+        Assert that only not updated projects are returned when status is
+        set to 'newer_updated'.
+        """
+        create_project(self.session)
+        for project in self.session.query(models.Project).all():
```

Comment:
FYI ``models.Project.query.all()`` should be equivalent here (db/meta.py sets up the query property). You don't need to change it, but all the existing code passing the session around is from code I never refactored and isn't necessary.